app/src/main/java/ch/epfl/polybazaar/UserProfileActivity.java

```diff
@@ -102,4 +105,19 @@ else if (!Utilities.passwordIsValid(newPassword)){
             });
         }
     }
+
+    public void viewOwnListings(View view) {
+        AppUser user = getUser();
```

Comment:
Since this method is used in UserProfileActivity, there must be a logged user if it is called. However I added the additional check for more robustness.